src/M6Web/Component/RedisMock/RedisMock.php

```diff
@@ -896,6 +896,14 @@ protected function zrangebyscoreHelper($key, $min, $max, array $options = array(
             $options['limit'] = array(0, count(self::$dataValues[$this->storage][$key]));
         }
 
+        $castResultValueAsString = function ($array) {
+            $result = [];
+            foreach ($array as $key => $value) {
+                $result[$key] = (string) $value;
+            }
+            return $result;
+        };
```

Comment:
You don't need an anonymous function for this, you can safely combine [`array_map`](https://www.php.net/manual/en/function.array-map.php) and [`strval`](https://www.php.net/manual/en/function.strval.php).
Then, in lines below you can write:
```php
return $this->returnPipedInfo(array_map('strval', $slice));
```

Reply:
We need to have the key of the array.

Reply:
This behavior is clearly described in documentation (at the end of the [4th example](https://www.php.net/manual/en/function.array-map.php#example-6224) ):
> The returned array will preserve the keys of the array argument if and only if exactly one array is passed. If more than one array is passed, the returned array will have sequential integer keys.

So in this case, with only one array, it will keep input keys. And you are right, with more arrays you have to combine yourself `array_map` result with expected keys.

Reply:
Damn, I was wrong about array_map the whole time ! Thanks for the input. I will push another version :)